modules/core/shared/src/main/scala/retry/PolicyDecision.scala

```diff
@@ -1,13 +1,8 @@
 package retry
 
 import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.DurationInt
 
-sealed trait PolicyDecision
-
-object PolicyDecision {
-  case object GiveUp extends PolicyDecision
-
-  final case class DelayAndRetry(
-      delay: FiniteDuration
-  ) extends PolicyDecision
-}
+enum PolicyDecision(val delay: FiniteDuration):
```

Comment:
We don't need to add this `delay` param to the enum, do we? I think this should be fine:

```scala
enum PolicyDecision:
  case GiveUp
  case DelayAndRetry(delay: FiniteDuration)